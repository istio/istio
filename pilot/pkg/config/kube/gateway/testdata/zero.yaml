apiVersion: gateway.networking.k8s.io/v1
kind: GatewayClass
metadata:
  name: istio
spec:
  controllerName: istio.io/gateway-controller
---
apiVersion: gateway.networking.k8s.io/v1
kind: Gateway
metadata:
  name: gateway
  namespace: istio-system
spec:
  addresses:
  - value: istio-ingressgateway
    type: Hostname
  gatewayClassName: istio
  listeners:
  - name: default
    hostname: "*.domain.example"
    port: 80
    protocol: HTTP
    allowedRoutes:
      namespaces:
        from: All
  - name: tcp
    port: 34000
    protocol: TCP
    allowedRoutes:
      namespaces:
        from: All
---
apiVersion: gateway.networking.k8s.io/v1
kind: HTTPRoute
metadata:
  name: http
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
  hostnames: ["first.domain.example"]
  rules:
  - matches:
    - path:
        type: PathPrefix
        value: /get
    backendRefs:
      - name: httpbin-zero
        port: 8080
        weight: 0
  - matches:
    - path:
        type: PathPrefix
        value: /weighted-100
    backendRefs:
    - filters:
      - requestHeaderModifier:
          add:
          - name: foo
            value: bar
        type: RequestHeaderModifier
      port: 8000
      name: foo-svc
      weight: 100
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: TCPRoute
metadata:
  name: tcp
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
  rules:
  - backendRefs:
      - name: httpbin-zero
        port: 8080
        weight: 0
