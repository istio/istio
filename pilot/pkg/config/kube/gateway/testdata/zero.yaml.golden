apiVersion: networking.istio.io/v1
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-semantics: gateway
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parents: Gateway/gateway/default.istio-system
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway-default
  namespace: istio-system
spec:
  servers:
  - hosts:
    - '*/*.domain.example'
    port:
      name: default
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-semantics: gateway
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parents: Gateway/gateway/tcp.istio-system
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway-tcp
  namespace: istio-system
spec:
  servers:
  - hosts:
    - '*/*'
    port:
      name: default
      number: 34000
      protocol: TCP
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/http.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: istio-system~gateway-istio-autogenerated-k8s-gateway-default~first.domain.example
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-default
  hosts:
  - first.domain.example
  http:
  - match:
    - uri:
        prefix: /weighted-100
    name: default.http.1
    route:
    - destination:
        host: foo-svc.default.svc.domain.suffix
        port:
          number: 8000
      headers:
        request:
          add:
            foo: bar
  - directResponse:
      status: 500
    match:
    - uri:
        prefix: /get
    name: default.http.0
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: TCPRoute/tcp.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: tcp-tcp-0-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-tcp
  hosts:
  - '*'
  tcp:
  - route:
    - destination:
        host: internal.cluster.local
        port:
          number: 65535
        subset: zero-weight
---
